### DISCUSSION TITLE
How do I mock useShallow

### DISCUSSION THREAD

**Discussion by shahbazshueb** [pmndrs/zustand#2100] [Q&A]
I have started using useShallow but now my tests are failing. How do I mock useShallow?

**Comment by dbritto-dev** [comment]
> I have started using useShallow but now my tests are failing. How do I mock useShallow?

@shahbazshueb you shouldn't mock `useShallow`. BTW, would you mind sharing us a codesandbox demo with a minimal repro? I would like to help

**Reply by shahbazshueb** [reply]
Sure. Will provide you with a codesandbox demo later today but I am getting the following error when running the tests:
```
TypeError: (0 , _shallow.useShallow) is not a function
```

**Reply by shahbazshueb** [reply]
Not sure what the issue is but for now, I copied over useShallow code in my codebase and started using it and the tests are passing now.
Used the following code:

```
import { useRef } from "react";
import { shallow } from "zustand/shallow";

//ToDO: Replace with zustand's useShallow after test issue fixed
export function useShallow<S, U>(selector: (state: S) => U): (state: S) => U {
  const prev = useRef<U>();

  return (state) => {
    const next = selector(state);
    return shallow(prev.current, next)
      ? (prev.current as U)
      : (prev.current = next);
  };
}
```

**Reply by dbritto-dev** [reply]
@shahbazshueb just share me a codesandbox demo so I can help you
